### ISSUE TITLE
Bot request: startblock

### ISSUE THREAD

**Issue by jmdobry** [googleapis/repo-automation-bots#150]
Create a Probot version of Startblock

**Comment by orthros** [comment]
Startblock is a tool the DRIFT team has used when managing PRs/CLs. 

The tl;dr is IFTTT for Pull Requests and code review.

Consider a PR that has been made with this snippet in the body of the PR

```
Startblock:
  has lgtm from orthros
  has lgtm from jmdobry
  has label good-to-merge
  ci passes
```

The bot would then iterate through these gated steps and as soon as all conditions pass, would merge the PR.

The bot can, however have more complex logic put into it in order to "gate" certain conditions, and modify the PR itself so as to not spam users.

Consider instead this Startblock declaration
```
Startblock:
  has lgtm from orthros
  and then
  add reviewer jmdobry
  and then
  has lgtm from jmdobry
  and then
  add label kokoro:run
  and then 
  has label good-to-merge
  ci passes
  and is Sunday
```

This declaration tells the bot to not request review from `jmdobry` until after `orthros` has given an lgtm, which keeps `jmdobry` from having to review changes that are unready for them to view.

**Comment by jmdobry** [comment]
The bot could manage its state on a PR via review comments. When the bot is first added as a reviewer, it parses the PR body for the bot conditions, then adds a review: Approval if the conditions have all been met or Changes Requested if something in the condition chain is unmet. The comment itself contains a prettier version of the condition block, describing the bot's idea of what needs to happen. The bot re-examines the PR and conditions continuously (some of the conditions might be time-based).

**Comment by tmatsuo** [comment]
Github has it's own way of configure branch protection rules, and I think most of the repos are using it.
Probably it's very unpopular feature request, and will never be implemented unless @jmdobry write it.

Closing, but feel free to reopen.

**Comment by jmdobry** [comment]
This feature request was not describing a form of branch protection, rather it describes a 100x more useful/powerful version of your merge on green bot.

**Comment by tmatsuo** [comment]
@jmdobry That's nice! Feel free to contribute or build your own :)

**Comment by tmatsuo** [comment]
@jmdobry 
To be clear, I agree this is a cool idea, and it is useful, but considering the current situation, this feature request won't be prioritized in our team before other things.

**Comment by bcoe** [comment]
Going to go ahead and reopen this, because @danielbankhead and I have been talking about a feature that is a subset of this functionality. For purposes of enforcing SLSA code review rules. I think it might be worth thinking about some useful features from Startblock while we work on this automation.

**Comment by tmatsuo** [comment]
I guess @danielbankhead went a different route with allstar.

**Comment by danielbankhead** [comment]
@tmatsuo I think we can keep this open - the Allstar Review bot covers a subset of this functionality, but as expressive as this feature request

**Comment by tmatsuo** [comment]
Thanks! I'd close this as wontfix
